OSDOCS-15763:Update the z-stream RNs for 4.18.22 (#97456)
openshift/openshift-docs: release_notes/ocp-4-18-release-notes.adoc

** You are upgrading from 4.17 to 4.18 during an update of the OVN-Kubernetes image.
** During the upgrade, if a pod on another system that was selected by an egress IP was deleted when the `ovnkube-node` pod was not running. (link:https://issues.redhat.com/browse/OCPBUGS-59531[OCPBUGS-59531])

[id="ocp-4-18-22-bug-fixes_{context}"]
==== Bug fixes

* Before this update, destroying a cluster in the unsupported region `mx-central-1` caused the destroyer to fail to find a partition and not exit. As a consequence, you could not destroy an {product-title} cluster in the `mx-central-1`region due to constant error reporting. With this release, the destroyer does not report errors for the unsupported region `mx-central-1`, which enables the successful destruction of a cluster. (link:https://issues.redhat.com/browse/OCPBUGS-56177[OCPBUGS-56177])

* Before this update, combined specification and status updates lists triggered unnecessary firmware upgrades, which caused system downtime. With this release, a firmware upgrade optimization skips unnecessary firmware upgrades. (link:https://issues.redhat.com/browse/OCPBUGS-56766[OCPBUGS-56766])

> * Before this update, the `console-telemetry` plugin received a `Forbidden` error due to using the wrong API endpoint for tracking usage. As a consequence, the `Forbidden error in console-telemetry-plugin usage tracking` error occurred. With this release, the `console-telemetry` plugin posts usage data to `/api/metrics/usage` instead of `/metrics/usage`. As a result, the `console-telemetry`plugin does not receive a `Forbidden` error, which ensures accurate usage tracking. (link:https://issues.redhat.com/browse/OCPBUGS-58364[OCPBUGS-58364])

Reviewer: This seems off:
> the `Forbidden error in console-telemetry-plugin usage tracking` error

Is that a paste or formatting issue? Or is the error really called "Forbidden error in console-telemetry-plugin usage tracking"?

Reviewer: also missing space in here is causing rendering issues
> the `console-telemetry`plugin does not

> * Before this update, the installer failed when {aws-first} credentials were not found and the survey was attempting to list all {aws-short} regions preventing users from creating the install-config. With this release, the installer no longer fails when {aws-short} credentials are not set, allow users to input them during the survey. (link:https://issues.redhat.com/browse/OCPBUGS-59155[OCPBUGS-59155])

Reviewer: ```suggestion
* Before this update, the installation program failed when {aws-first} credentials were not found and the survey was attempting to list all {aws-short} regions preventing users from creating the `install-config` YAML file. With this release, the installation program no longer fails when {aws-short} credentials are not set, allow users to input them during the survey. (link:https://issues.redhat.com/browse/OCPBUGS-59155[OCPBUGS-59155])
```

> * Before this update, when a `HostedCluster` was configured with a proxy URL such as http://user:pass@host, the authentication header was not getting forwarded by the konnectivity proxy to the user proxy, failing authentication. With this release, the proper authentication header is sent when a user and password is specified in the proxy URL. (link:https://issues.redhat.com/browse/OCPBUGS-59503[OCPBUGS-59503])

Reviewer: Items in backticks need a noun to follow. Can this just be "hosted cluster" to simplify that? 

You also need to escape the URL so it's not clickable text.

> * Before this update, the `oc-mirror`did not detect Helm Chart images that used an aliased sub-chart. As a consequence, the Helm Chart images were missing after mirroring. With this release, the `oc-mirror`detects and mirrors Helm Chart images with an aliased sub-chart. (link:https://issues.redhat.com/browse/OCPBUGS-59798[OCPBUGS-59798])

Reviewer: ```suggestion
* Before this update, the `oc-mirror` did not detect Helm Chart images that used an aliased sub-chart. As a consequence, the Helm Chart images were missing after mirroring. With this release, the `oc-mirror` detects and mirrors Helm Chart images with an aliased sub-chart. (link:https://issues.redhat.com/browse/OCPBUGS-59798[OCPBUGS-59798])
```
Check your output on backticks especially to make sure they are rendering how they should 
<img width="658" height="111" alt="image" src="https://github.com/user-attachments/assets/8df9a40e-6b13-4a4e-999d-d1b22474fb43" />

> * Before this update, worker scale-up jobs that used Podman v5 with the `container-tools` module for `netavark` failed due to denial of the Open Container Initiative (OCI) permission when writing `devices.allow` with `crun`. As a consequence, the container scaling jobs failed. With this release, the `netavark` dependency for Podman is disabled, and `runc` runtime is used instead. As a result, worker scale-up jobs are now successful. (link:https://issues.redhat.com/browse/OCPBUGS-59843[OCPBUGS-59843])

Reviewer: We don't use "worker" unless it is a code literal (i.e. `role = worker` in a config file), these should likely be "compute" instead

> * Before this update, when you cloned a .tar file with zero length, the `oc-mirror` ran indefinitely due to an empty archive file. As a consequence, no progress occurred when you mirrored a 0-byte .tar file. With this release, 0-byte tar files are detected and reported as errors, which prevents the `oc-mirror` from hanging. (link:https://issues.redhat.com/browse/OCPBUGS-59864[OCPBUGS-59864])

Reviewer: either put the .tar extension in backticks or change to TAR

> * Before this update, in multi-zone clusters with only a single worker per zone, if the Monitoring Operator's Prometheus pods were scheduled to nodes that reboot back-to-back and both reboots took longer than 15 minutes to return to service, the Monitoring Operator might have degraded. With this release, the time-out has been extended to 20 minutes to prevent the Monitoring Operator from entering a degraded state on common cluster topologies. Clusters where the two nodes with Prometheus pods reboot back-to-back and take more than 20 minutes might still report a degraded state until the second node and Prometheus pod return to a normal state.(link:https://issues.redhat.com/browse/OCPBUGS-59962[OCPBUGS-59962])

Reviewer: worker should be compute here too

* Before this update, when you ran the oc-mirror v2 disk-to-mirror workflow without valid mirror tar files, the returned error messages did not correctly identify the problem. With this release, the oc-mirror v2 workflow returns an error message that states `no tar archives matching "mirror_[0-9]{6}\.tar" found in "<directory>"`. (link:https://issues.redhat.com/browse/OCPBUGS-59235[OCPBUGS-59235])

> * Before this update, when a Machine Set was scaled down and had reached its minimum size, the Cluster Autoscaler could leave the last remaining node with a no schedule taint that prevented use of a node. This issue was caused by a counting error in the Cluster Autoscaler. With this release, the counting error has been fixed so that the Cluster Autoscaler works as expected when a Machine Set is scaled down and has reached its minimum size. (link:https://issues.redhat.com/browse/OCPBUGS-59260[OCPBUGS-59260])

Reviewer: Existing content but FYI
```suggestion
* Before this update, when a machine set was scaled down and had reached its minimum size, the Cluster Autoscaler could leave the last remaining node with a no schedule taint that prevented use of a node. This issue was caused by a counting error in the Cluster Autoscaler. With this release, the counting error has been fixed so that the Cluster Autoscaler works as expected when a machine set is scaled down and has reached its minimum size. (link:https://issues.redhat.com/browse/OCPBUGS-59260[OCPBUGS-59260])
```
